beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -209,15 +268,17 @@ func (vs *Server) eth1DataMajorityVote(ctx context.Context, beaconState state.Be
 	earliestValidTime := votingPeriodStartTime - 2*params.BeaconConfig().SecondsPerETH1Block*eth1FollowDistance
 	latestValidTime := votingPeriodStartTime - params.BeaconConfig().SecondsPerETH1Block*eth1FollowDistance
 
-	lastBlockByEarliestValidTime, err := vs.Eth1BlockFetcher.BlockByTimestamp(ctx, earliestValidTime)
-	if err != nil {
-		log.WithError(err).Error("Could not get last block by earliest valid time")
-		return vs.randomETH1DataVote(ctx)
-	}
-	// Increment the earliest block if the original block's time is before valid time.
-	// This is very likely to happen because BlockTimeByHeight returns the last block AT OR BEFORE the specified time.
-	if lastBlockByEarliestValidTime.Time < earliestValidTime {
-		lastBlockByEarliestValidTime.Number = big.NewInt(0).Add(lastBlockByEarliestValidTime.Number, big.NewInt(1))
+	if !featureconfig.Get().EnableGetBlockOptimizations {
```

Comment:
Why is this branch omitted when the flag is on?

Reply:
This branch seems to not do anything here.  
`lastBlockByEarliestValidTime` is never used anywhere inside the code.
I need a confirmation if my thinking is right.
WDYT @nisdas

Reply:
That is a bit more complicated, as that value was used for our `inRangeVotes` method. This was removed previously when we had the voting bug and defaulted to independent voting compared to voting by majority. Do we ever expect to add the vote by majority feature back ? @prestonvanloon

Reply:
My 2 cents: Even if want to add the feature back sometime later. Let's remove unused code for now.

Reply:
Discussed offline, its better to just remove it now. Performing the extra BlockByTimeStamp call is expensive and can increase block proposal times needlessly.